aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -706,6 +739,32 @@ impl VMChangeSet {
         Ok(())
     }
 
+    fn squash_additional_group_reads_needing_exchange<K>(
+        group_reads_needing_exchange: &mut BTreeMap<StateKey, (WriteOp, u64)>,
+        additional_group_reads_needing_exchange: BTreeMap<StateKey, (WriteOp, u64)>,
+        skip: &BTreeMap<StateKey, K>,
```

Comment:
This (and in `squash_additional_reads_needing_exchange`) can be just a set, not a map?

Reply:
but then we would need to create a temporary set, vs just passing reference to the object we already have?

maybe there is a nicer way to say that, maybe through some lambda or something?

Reply:
can't find a clean solution, and seems we'd need some sort of keys as set interface
https://users.rust-lang.org/t/set-difference-on-btreeset-and-btreemap-keys/20111